src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.NativeAot.cs

```diff
@@ -1490,530 +71,128 @@ internal static unsafe uint IUnknown_Release(IntPtr pThis)
-        //                    1. We can either create a new NoCW or get back an old one from the cache.
-        //                    2. This obj could be a regular tracker runtime object for data binding.
-        //  ppNewReference  - The IReferenceTrackerTarget* for the proxy created
-        //                    The tracker runtime will call IReferenceTrackerTarget to establish a reference.
-        //
-        [UnmanagedCallersOnly]
-        internal static unsafe int IReferenceTrackerHost_GetTrackerTarget(IntPtr pThis, IntPtr punk, IntPtr* ppNewReference)
-        {
-            if (punk == IntPtr.Zero)
-            {
-                return HResults.E_INVALIDARG;
-            }
-
-            if (Marshal.QueryInterface(punk, IID_IUnknown, out IntPtr ppv) != HResults.S_OK)
-            {
-                return HResults.COR_E_INVALIDCAST;
-            }
-
-            try
-            {
-                using ComHolder identity = new ComHolder(ppv);
-                using ComHolder trackerTarget = new ComHolder(GetOrCreateTrackerTarget(identity.Ptr));
-                return Marshal.QueryInterface(trackerTarget.Ptr, IID_IReferenceTrackerTarget, out *ppNewReference);
-            }
-            catch (Exception e)
-            {
-                return Marshal.GetHRForException(e);
-            }
-        }
-
-        [UnmanagedCallersOnly]
-        internal static unsafe int IReferenceTrackerHost_AddMemoryPressure(IntPtr pThis, long bytesAllocated)
-        {
-            try
-            {
-                GC.AddMemoryPressure(bytesAllocated);
-                return HResults.S_OK;
-            }
-            catch (Exception e)
-            {
-                return Marshal.GetHRForException(e);
-            }
-        }
-
-        [UnmanagedCallersOnly]
-        internal static unsafe int IReferenceTrackerHost_RemoveMemoryPressure(IntPtr pThis, long bytesAllocated)
+        private static IntPtr GetTaggedImplCurrentVersion()
         {
-            try
-            {
-                GC.RemoveMemoryPressure(bytesAllocated);
-                return HResults.S_OK;
-            }
-            catch (Exception e)
+            unsafe
             {
-                return Marshal.GetHRForException(e);
+                return (IntPtr)(delegate* unmanaged<IntPtr, IntPtr, int>)&VtableImplementations.ITaggedImpl_IsCurrentVersion;
             }
         }
 
-        // Lifetime maintained by stack - we don't care about ref counts
-        [UnmanagedCallersOnly]
-        internal static unsafe uint Untracked_AddRef(IntPtr pThis)
-        {
-            return 1;
-        }
-
-        [UnmanagedCallersOnly]
-        internal static unsafe uint Untracked_Release(IntPtr pThis)
-        {
-            return 1;
-        }
-
-        [UnmanagedCallersOnly]
-        internal static unsafe int IReferenceTrackerHost_QueryInterface(IntPtr pThis, Guid* guid, IntPtr* ppObject)
+        private static unsafe IntPtr CreateDefaultIUnknownVftbl()
```

Comment:
```suggestion
        internal static IntPtr DefaultIUnknownVftblPtr
            => (IntPtr)Unsafe.AsPointer(in VtableImplementations.IUnknown);
```
And move DefaultIUnknownVftblPtr impl that does caching into CoreCLR specific part. Caching the vtable addressed is de-optimization.

Same for CreateTaggedImplVftbl and CreateDefaultIReferenceTrackerTargetVftbl.